src/poll.rs

```diff
@@ -68,9 +82,17 @@ impl PollFd {
     }
 }
 
-impl AsRawFd for PollFd {
-    fn as_raw_fd(&self) -> RawFd {
-        self.pollfd.fd
+impl AsFd for PollFd {
+    fn as_fd(&self) -> BorrowedFd<'_> {
+        // BorrowedFd::borrow_raw() requires that the raw fd being passed must
+        // remain open for the duration of the returned BorrowedFd, this is
+        // guaranteed as the returned BorrowedFd has lifetime parameter same
+        // as `self`:
+        // "fn as_fd<'self>(&'self self) -> BorrowedFd<'self>"
+        // which means that `self` (PollFd) is guaranteed to outlive the returned
+        // BorrowedFd. Since `self` is valid, the fd within `self` is valid,
```

Comment:
Not quite.  As written there's no way we can guarantee that self is valid.  For example
```rust
let f = tempfile();
let p = PollFd::new(&f, ...);
drop(f);
let _ = p.as_fd();    // Invalid fd!
```